wagtail/wagtailadmin/views/pages.py

```diff
@@ -629,8 +629,17 @@ def unpublish(request, page_id):
     next_url = get_valid_next_url_from_request(request)
 
     if request.method == 'POST':
+        include_descendants = request.POST.get("include_descendants", False)
+
         page.unpublish()
 
+        if include_descendants == 'on':
+            live_descendant_pages = page.get_descendants().live()
+            for live_descendant_page in live_descendant_pages:
+                live_descendant_page = live_descendant_page.specific
+                if live_descendant_page.permissions_for_user(request.user).can_unpublish():
+                    live_descendant_page.unpublish()
```

Comment:
Considering that not being able to unpublish the root page raises a `PermissionDenied` error above, silently ignoring this issue here is surprising. In my opinion, either the whole tree should be unpublished, or nothing should be unpublished. Perhaps, if the include_decendants checkbox is checked, the descendants shoul all be checked for permission before unpublishing them, and an error being raised if the user does not have permission.